### PR TITLE
ref(js): Simplify TeamBadge component

### PR DESCRIPTION
**Pull request by evanpurkhiser**
AFAICT this did not need to have all this crazy logic to just get the
most up-to-date team from the store

Remvoes an instance of Store.listen

### CODE REVIEW

**Comment by evanpurkhiser** [comment]
Something is wonkily different about this.

**Comment by scttcper** [comment]
I think it did need that logic

File: static/app/views/settings/organizationTeams/teamSettings/index.tsx [hunk 1]
```diff
@@ -38,7 +38,9 @@ class TeamSettings extends AsyncView<Props, State> {
   }
 
   handleSubmitSuccess = (resp: any, model: FormModel, id?: string) => {
-    updateTeamSuccess(resp.slug, resp);
+    // Use the old slug when triggering the update so we correctly replace the
+    // previous team in the store
+    updateTeamSuccess(this.props.team.slug, resp);
```
Comment (evanpurkhiser) [inline static/app/views/settings/organizationTeams/teamSettings/index.tsx:43]: This problem was exposed by this change

Comment (JonasBa) [reply static/app/views/settings/organizationTeams/teamSettings/index.tsx:43]: Damn, nice find

**Review by scttcper — APPROVED** [review]

**Pull request merged**